v6/info/features.go

```diff
@@ -16,15 +16,40 @@ type SupportedFeatures struct {
 	TouchGenerationsTitle  uint8
 }
 
-func (i *Info) GetSupportedFeatures(features *gametdb.Features) {
+func (i *Info) GetSupportedFeatures(features *gametdb.Features, gameID string, controllers *gametdb.Controllers) {
+	touchGenIDs := []string{
+		"YBN", "VAA", "AYA", "AND", "ANM", "ATD", "CVN",
+		"YCU", "ATI", "AOS", "AG3", "AWI", "APL", "AJQ", "CM7",
+		"AD5", "AD2", "ADG", "AD7", "AD3", "IMW", "C6P", "AXP",
+		"A8N", "AZI", "ASQ", "ATR", "AGF",
+		"RFN", "RFP", "R64", "RYW",
+	}
+
 	for _, s := range features.Feature {
 		if strings.Contains(s, "online") {
 			i.SupportedFeatures.Online = 1
 			i.SupportedFeatures.NintendoWifiConnection = 1
-		} else if s == "nintendods" {
-			i.SupportedFeatures.DownloadPlay = 1
 		} else if s == "download" {
 			i.SupportedFeatures.DLC = 1
 		}
 	}
+
+	if controllers.MultiCart >= 2 && controllers.MultiCart <= 8 {
+		i.SupportedFeatures.WirelessPlay = 1
+	}
+	if controllers.SingleCart >= 1 && controllers.SingleCart <= 8 {
+		i.SupportedFeatures.DownloadPlay = 1
+	}
```

Comment:
I'm pretty sure it should be `controllers.SingleCart > 1`. I checked Phoenix Wright: Ace Attorney Justice for All which does not have Download Play, however this logic says it does.

Also the upper bounds check is redundant.